File: internal/parser/testdata/sample.rb
# A sample Ruby file for testing the parser

require 'json'
require_relative 'helper'

# The User class handles user operations.
# It extends BaseModel for persistence.
class User < BaseModel
  include Printable
  extend ClassMethods

  attr_reader :name, :email
  attr_accessor :age

  ROLE_ADMIN = 'admin'
  MAX_AGE = 150

  # Creates a new user instance.
  def initialize(name, email, age)
    @name = name
    @email = email
    @age = age
  end

  # Returns a greeting string for the user.
  def greet(greeting)
    "#{greeting}, #{@name}!"
  end

  # Finds a user by ID.
  def self.find(id)
    # class method lookup
  end

  def self.create(attrs)
    # another class method
  end

  private

  def validate_email
    raise "Invalid" if @email.nil?
  end

  def test_greeting
    # test method
  end

  def test_validation
    # another test
  end
end

# Serializable module for JSON conversion.
module Serializable
  def to_json
    # serialize to JSON
  end

  def from_json(data)
    # deserialize from JSON
  end
end

File: internal/parser/testdata/sample.lua
local json = require("json")
local utils = require("utils")

--- A sample user class.
--- Handles user operations.
local User = {}
User.__index = User

--- Creates a new user instance.
function User:new(name, age)
  local self = setmetatable({}, User)
  self.name = name
  self.age = age
  return self
end

--- Returns a greeting string.
function User:greet(greeting)
  return greeting .. ", " .. self.name .. "!"
end

function User.find(id)
  return nil
end

local function helper()
  return 42
end

function globalFunc(x)
  return x * 2
end

function test_user_creation()
  -- test
end

function test_greeting()
  -- test
end
